codes/quantum/properties/symmetry_protected.yml

```diff
@@ -0,0 +1,31 @@
+code_id: symmetry_protected
```

Comment:
- Please add a stub for "Self-correcting quantum code" (code_id: self_correct). 
- Add hamiltonian as parent to self_correct; add higher_dimensional_surface as cousin, mention that it's a self-correcting memory in 4D.
- Add self_correct as parent to this entry; remove hamiltonian.